Signal/src/util/OWSScreenLockUI.m

```diff
@@ -116,19 +118,35 @@ - (void)setupWithRootWindow:(UIWindow *)rootWindow
 
 - (void)setAppIsInactive:(BOOL)appIsInactive
 {
+    BOOL didChange = _appIsInactive != appIsInactive;
+
     _appIsInactive = appIsInactive;
 
+    if (didChange) {
+        // If app is inactive for more than N seconds,
+        // treat this as "entering the background" for the purposes
+        // of Screen Lock.
+        if (!appIsInactive) {
+            [self.inactiveTimer invalidate];
+            self.inactiveTimer = nil;
+        } else if (!self.isShowingScreenLockUI) {
+            [self.inactiveTimer invalidate];
+            self.inactiveTimer = [NSTimer weakScheduledTimerWithTimeInterval:45.f
```

Comment:
Is this to handle the case when the app is in the app switcher for a long time?

Reply:
Yes.